text/0000-ancillary-package-module-standardization.md

```diff
@@ -0,0 +1,240 @@
+
+### Under the Hood
+
+We use two tools to produce the desired output formats:
+
+1. The TypeScript compiler, which converts TypeScript and ES2015+ to
+   ES5, in both CommonJS and standard module syntax.
+2. Rollup, to package the ES5 source into legacy bundle formats like
+   UMD.
+
+## Examples
+
+| Repo                 | Description                                                                                               | PR                                                 |
+|----------------------|-----------------------------------------------------------------------------------------------------------|----------------------------------------------------|
+| [fastboot](https://github.com/ember-fastboot/fastboot) (Node-only) | Uses broccoli-module-alchemist to utilize Node 4's ES2015 syntax features while remaining 0.12 compatible | https://github.com/ember-fastboot/fastboot/pull/74 |
+
+# How We Teach This
+
+Because this is an internal infrastructure concern, it does not have
+user-facing impact. That said, we should ensure that ease-of-use and
+maintainability are priorities to enable new contributors to get
+started.
+
+Because users who need this information are intermediate to advanced,
+this RFC itself should serve as a good starting point. Additionally,
+configuration information and how-tos will be incorporated into
+`broccoli-module-alchemist`'s README.
+
+# Drawbacks
+
+The biggest drawback is that this change may introduce temporary
+instability into the ecosystem. Many packages may rely on non-standard
+semantics or syntax, and updating them may divert work that could go
+elsewhere.
+
+Additionally, this RFC does not go so far as attempting to standardize
+the testing infrastructure. Some packages use mocha, while others use
+QUnit. Some run their tests in Node.js, others in PhantomJS. Ideally, a
+future RFC could attempt this unification; although rewriting entire
+test suites just to use a different test harness does not seem like the
+best use of precious time and energy.
+
+# Alternatives
+
+The obvious alternative is to do nothing; the current system works,
+although it does preclude several nice optimizations. In terms of
+alternate module systems, there does not seem to be a better subset than
+JavaScript modules. JavaScript modules are the only standard to be
+statically analyzable; it is much more difficult to translate from a
+runtime-dependent format (like CommonJS) into a static format (like JS
+modules) than vice versa.
+
+# Unresolved questions
+
+- What is the exact file system layout?
+- How do we handle "multi-package" packages, like Glimmer, which include
+  multiple logical packages in a `packages` directory? Should each
+  package have its own `package.json`?
+- Does it make sense to TypeScript-enable the infrastructure, given that
+  many new ecosystem packages rely on it?
+- Is it possible to conventionalize testing? If the only two systems in
+  common use are Mocha and QUnit, can we support both?
```

Comment:
@chadhietala you should jump in, and just make sure what happens here doesn't prevent or hurt the future of linker/ etc.